cmd/podman/root.go

```diff
@@ -109,8 +109,40 @@ func persistentPreRunE(cmd *cobra.Command, args []string) error {
 		return nil
 	}
 
+	// Work around https://github.com/spf13/cobra/issues/412
+	_ = cmd.LocalFlags()
+
 	cfg := registry.PodmanConfig()
 
+	// --connection is not as "special" as --remote so we can wait and process it here
+	var err error
+	// override --url and --identity flags if --connection given
+	cmd.VisitParents(func(cmd *cobra.Command) {
```

Comment:
`cmd.Root().LocalFlags()` should give you the flags you need without the work around and without having to loop

Reply:
@Luap99 Thank you for the suggestion!